pkg/server/api/svid/v1/service_test.go

```diff
@@ -0,0 +1,246 @@
+package svid_test
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/spire/pkg/server/api/rpccontext"
+	"github.com/spiffe/spire/pkg/server/api/svid/v1"
+	"github.com/spiffe/spire/pkg/server/ca"
+	"github.com/spiffe/spire/test/clock"
+	"github.com/spiffe/spire/test/fakes/fakeserverca"
+	"github.com/spiffe/spire/test/spiretest"
+	"github.com/spiffe/spire/test/testkey"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+)
+
+func TestServiceMintX509SVID(t *testing.T) {
+	clock := clock.NewMock(t)
```

Comment:
nit: Unless you need to control the clock, you can just omit this option and the `fakeserverca` will create it's own (that you can get at via the `Clock()` method like you are below.